invokeai/backend/tiles/tiles.py

```diff
@@ -63,31 +97,122 @@ def calc_tiles_with_overlap(
 
             tiles.append(tile)
 
-    def get_tile_or_none(idx_y: int, idx_x: int) -> Union[Tile, None]:
-        if idx_y < 0 or idx_y > num_tiles_y or idx_x < 0 or idx_x > num_tiles_x:
-            return None
-        return tiles[idx_y * num_tiles_x + idx_x]
+    return calc_overlap(tiles, num_tiles_x, num_tiles_y)
+
+
+def calc_tiles_even_split(
+    image_height: int, image_width: int, num_tiles_x: int, num_tiles_y: int, overlap: float = 0
+) -> list[Tile]:
+    """Calculate the tile coordinates for a given image shape with the number of tiles requested.
+
+    Args:
+        image_height (int): The image height in px.
+        image_width (int): The image width in px.
+        num_x_tiles (int): The number of tile to split the image into on the X-axis.
+        num_y_tiles (int): The number of tile to split the image into on the Y-axis.
+        overlap (int, optional): The target overlap amount of the tiles size. Defaults to 0.
+
+    Returns:
+        list[Tile]: A list of tiles that cover the image shape. Ordered from left-to-right, top-to-bottom.
+    """
+
+    # Ensure tile size is divisible by 8
+    if image_width % 8 != 0 or image_height % 8 != 0:
+        raise ValueError(f"image size (({image_width}, {image_height})) must be divisible by 8")
+
+    # Calculate the overlap size based on the percentage and adjust it to be divisible by 8 (rounding up)
+    overlap_x = 8 * math.ceil(int((image_width / num_tiles_x) * overlap) / 8)
+    overlap_y = 8 * math.ceil(int((image_height / num_tiles_y) * overlap) / 8)
+
+    # Calculate the tile size based on the number of tiles and overlap, and ensure it's divisible by 8 (rounding down)
+    tile_size_x = 8 * math.floor(((image_width + overlap_x * (num_tiles_x - 1)) // num_tiles_x) / 8)
+    tile_size_y = 8 * math.floor(((image_height + overlap_y * (num_tiles_y - 1)) // num_tiles_y) / 8)
 
-    # Iterate over tiles again and calculate overlaps.
+    # tiles[y * num_tiles_x + x] is the tile for the y'th row, x'th column.
+    tiles: list[Tile] = []
+
+    # Calculate tile coordinates. (Ignore overlap values for now.)
     for tile_idx_y in range(num_tiles_y):
+        # Calculate the top and bottom of the row
+        top = tile_idx_y * (tile_size_y - overlap_y)
+        bottom = min(top + tile_size_y, image_height)
+        # For the last row adjust bottom to be the height of the image
+        if tile_idx_y == num_tiles_y - 1:
+            bottom = image_height
```

Comment:
I don't think it is as simple as getting "the math right", as the maths is fairly fairly basic and already correct in my opinion.  However, because of the "ensuring divisible by 8 parts" in the rounding up of the overlap size and rounding down of the tile sizes we can loose or gain some pixels. Doing this final step just adjusts the last tile to acount for this without having to track every loss or gain.  Personally I don't see much point in changing the behaviour of this part.  

However If you think the maths is flawed in some way or detrimental I am more than happy to be corrected on my opionion.

Reply:
Didn't notice any flaws, it just seemed like this final check might be redundant (though I haven't exercised all cases to be sure). If fine with keeping it if it doesn't bother you.